### ISSUE TITLE
Distance grabbing from a snap zone does not transition

### ISSUE THREAD

**Issue by thestonefox** [ExtendRealityLtd/Tilia.Interactions.PointerInteractors.Unity#226]
If you try and distance grab an object that is snapped in a snap zone then it doesn't transition to the interactor, instead it waits the transition time then instantly snaps.

This is because the object is still snapped into the snap zone until the transition has ended and then the grab happens.

One workaround is to add a `SnapZoneManager` to the internals of the distance grabber and have it unsnap before the grab takes place.

![image](https://user-images.githubusercontent.com/1029673/166436295-a64cfefe-93bc-4dbf-a63a-3257199ce5ee.png)

At present, the PointerInteractors do not have a dependency on the Snap Zone and I'm not sure if forcing that dependency is the best solution long term, so I'll leave this as a workaround for now until I come across a better solution

**Comment by fight4dream** [comment]
in the pic it has a notify before grabbed, meaning it has a “before grabbed” event on the facade?
then the script dont need to be inside the distance grabber.

**Comment by thestonefox** [comment]
yeah the script doesn't need to be in the distance grabber. but i kinda feel the distance grabber should _just work_ with a snap zone without a user having to set anything up, but then on the other hand i dont like the idea the distance grabber needs to know about snap zones

**Comment by fight4dream** [comment]
pattern-wise, if two parts dont need to know each other, then it is the work of a mediator. or messaging. 

i think the snapped interactable may be the key here. because it is snapped by the snapzone, the snapzone know of it. also the distance grabber want to grab it, the distance grabber know of it. 
they snapzone and grabber dont know each other, but the interactable can send message saying i am being requested. 

on the other hand, i feel like a snapzone is like an interactor. i could  use an interactor in place of a snapzone if i dont care about the “highlight”. Then i am curious that, if the distance grabber have the same issue when it attempt to snatch it from another interactor
